common/crypto/openssl/cert.c

```diff
@@ -1013,3 +1028,396 @@ oe_result_t oe_cert_get_validity_dates(
 
     return result;
 }
+
+#ifdef OE_BUILD_ENCLAVE
+/*
+ * Parse the name string into X509_NAME struct. The format of the string is
+ * "KEY1=VALUE1,KEY2=VALUE2,KEY3=VALUE3...".
+ */
+static X509_NAME* _X509_parse_name(const char* name_string)
```

Comment:
How locked down is this string format? As far as I can tell, the string comes directly from the user with no sanitation and no format checking ... does this need to be made more robust? (i.e. stripping spaces, checking for escape characters, etc.) It would be much preferable if we did not have to handle this directly.

Reply:
Updated the implementation based on mbedtls_x509_string_to_names and added a test.